packages/cactus-plugin-ledger-connector-corda/src/main/typescript/plugin-ledger-connector-corda.ts

```diff
@@ -243,4 +324,210 @@ export class PluginLedgerConnectorCorda
   public async getFlowList(): Promise<string[]> {
     return ["getFlowList()_NOT_IMPLEMENTED"];
   }
+
+  private async setupRequest(
+    username: string,
+    password: string,
+  ): Promise<{ headers: any; agent: any }> {
+    const authString = Buffer.from(`${username}:${password}`).toString(
+      "base64",
+    );
+    const headers = { Authorization: `Basic ${authString}` };
+    const rejectUnauthorized = false;
+    const httpsAgent = new https.Agent({
+      rejectUnauthorized,
+    });
+    return { headers, agent: httpsAgent };
+  }
+
+  private validateHoldingIDShortHash(inputString: string): boolean {
+    /* only hexadecimal characters and exactly 12 characters long as per the corda docs
+       https://docs.r3.com/en/platform/corda/5.2/developing-applications/cordapp-template/utxo-ledger-example-cordapp/running-the-chat-cordapp.html#using-swagger
+    */
+    const pattern = /^[0-9A-Fa-f]{12}$/;
+    return pattern.test(inputString);
+  }
+
+  public async startFlow(req: StartFlowV1Request): Promise<any> {
+    try {
+      const { headers, agent } = await this.setupRequest(
+        req.username,
+        req.password,
+      );
+      const holdingIDShortHash = req.holdingIDShortHash;
+      if (typeof holdingIDShortHash === "undefined") {
+        throw new Error("holdingIDShortHash is undefined");
+      }
+      if (!this.validateHoldingIDShortHash(holdingIDShortHash)) {
+        throw new Error(`Invalid holdingIDShortHash: ${holdingIDShortHash}`);
```

Comment:
@adrianbatuto Please throw a `BadRequestError` from this library https://www.npmjs.com/package/http-errors-enhanced-cjs instead of a regular `Error` it will help with the error handling higher up the callstack (the endpoints).